clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/securitygroup/SecurityGroupIngressConverter.groovy

```diff
@@ -130,4 +127,59 @@ class SecurityGroupIngressConverter {
       }
     }.flatten().unique()
   }
+
+  /**
+   *
+   * @param newList from description
+   * @param existingRules
+   * @return Map of rules that needs to be added , removed and updated
+   * Computes the delta between the existing rules and new rule
+   * Any rule present in description and not in the existing rule gets added to addition list.
+   * Any rule in description but present in existing rule get added to the remove list.
+   * Any rule with a change in description only gets added to the update list based on the following,
+   * - If a new rule has description value add it to update list to make it consistent.
+   * - If new rule has no description value set, ignore.
+   */
+  static Map<String, List> computeIpRuleDelta(List<IpPermission> newList, List<IpPermission> existingRules) {
```

Comment:
having this return Map<String, List> feels a bit brittle
I'd suggest making a concrete return type object 
```
@Canonical
static class IpRuleDelta {
  List<IpPermission> toAdd
  List<IpPermission> toRemove
  List<IpPermission> toUpdate
}
```

Reply:
good suggestion , updated to use a concrete object.